File: Leetcode/140125/Java/PrefixCommonArray.java
// Problem Link: https://leetcode.com/problems/find-the-prefix-common-array-of-two-arrays/?envType=daily-question&envId=2025-01-14

// Beginner friendly solution
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        
        Set<Integer> setA = new HashSet<>();
        Set<Integer> setB = new HashSet<>();
        int[] count = new int[A.length];

        for(int i=0; i<A.length; i++){

            setA.add(A[i]);
            setB.add(B[i]);

            int c=0;

            for(int ele: setA){

                if(setB.contains(ele))
                  c++;
            }
            count[i] = c;
        }

        return count;
        
    }
}

File: Leetcode/140125/Python/PrefixCommonArray.py
# Problem Link: https://leetcode.com/problems/find-the-prefix-common-array-of-two-arrays/?envType=daily-question&envId=2025-01-14
# Beginner Friendly Solution

class Solution(object):
    def findThePrefixCommonArray(self, A, B):
        """
        :type A: List[int]
        :type B: List[int]
        :rtype: List[int]
        """
        setA = set()
        setB = set()
        count = [0]*(len(A))

        for i in range(len(A)):
            setA.add(A[i])
            setB.add(B[i])

            for e in setA:
                if e in setB:
                   count[i]+=1

        return count
